kolibri/plugins/management/assets/src/device_management/test/views/channels-grid.spec.js

```diff
@@ -0,0 +1,146 @@
+    emptyState: () => wrapper.find('.no-channels'),
+    progressBar: () => wrapper.find(UiProgressLinear),
+    deleteChannelModal: () => wrapper.first(DeleteChannelModal),
+  };
+}
+
+describe('channelsGrid component', () => {
+  let store;
+
+  beforeEach(() => {
+    store = makeStore();
+    store.dispatch('SET_CHANNEL_LIST', [
+      {
+        name: 'visible channel',
+        id: 'visible_channel',
+        on_device_resources: 10,
+        total_resources: 1000,
+      },
+    ]);
+  });
+
+  it('shows an empty state if there are no visible channels', () => {
+    // "visible" meaning it has on-device resources
+    store.dispatch('SET_CHANNEL_LIST', [
+      {
+        name: 'hidden channel',
+        id: 'hidden_channel',
+        on_device_resources: 0,
+        total_resources: 1000,
+      },
+    ]);
+    const wrapper = makeWrapper({ store });
+    const { emptyState } = getElements(wrapper);
+    return wrapper.vm.$nextTick().then(() => {
+      assert(emptyState()[0].is('p'));
+    });
+  });
+
+  it('shows a progress bar if channels are loading', () => {
+    const wrapper = makeWrapper({ store });
+    const { progressBar } = getElements(wrapper);
+    return wrapper.vm
+      .$nextTick()
+      .then(() => {
+        wrapper.setData({ channelsLoading: true });
+        return wrapper.vm.$nextTick();
+      })
+      .then(() => {
+        assert(progressBar()[0].isVueComponent);
+      });
+  });
+
+  it('channels appear sorted by name', () => {
+    store.dispatch('SET_CHANNEL_LIST', [
+      {
+        name: 'beautiful channel',
+        id: 'beautiful_channel',
+        on_device_resources: 10,
+        total_resources: 1000,
+      },
+      {
+        name: 'awesome channel',
+        id: 'awesome_channel',
+        on_device_resources: 10,
+        total_resources: 1000,
+      },
+    ]);
+    const wrapper = makeWrapper({ store });
+    const { channelListItems } = getElements(wrapper);
+    return wrapper.vm.$nextTick().then(() => {
+      const [ch1, ch2] = channelListItems();
+      assert.equal(ch1.getProp('channel').id, 'awesome_channel');
+      assert.equal(ch2.getProp('channel').id, 'beautiful_channel');
+    });
+  });
+
+  it('a modal appears if channel is selected for deletion', () => {
+    // and clicking "confirm" triggers an action
+    let deleteModal;
+    const wrapper = makeWrapper({ store });
+    const deleteActionStub = sinon.stub(wrapper.vm, 'triggerChannelDeleteTask');
+    const { channelListItems, deleteChannelModal } = getElements(wrapper);
+    return wrapper.vm
+      .$nextTick()
+      .then(() => {
+        const [ch1] = channelListItems();
+        const button = ch1.first('button');
+        assert.equal(button.text().trim(), 'Delete');
+        button.trigger('click');
+        return wrapper.vm.$nextTick();
+      })
+      .then(() => {
+        deleteModal = deleteChannelModal();
+        assert(deleteModal.isVueComponent);
```

Comment:
What's the purpose of this assertion? Do you do this to all vue components?

Reply:
It's basically asserting that the deleteModal is mounted. If I don't care about any more details than that (e.g. its props or any other data), I will assert `.is('div')` for non vue-components, and `.isVueComponents` for vue components.

Reply:
I see, thanks for the clarification.